File: dongyimai-manager-web/src/main/webapp/js/service/TbItemCatService.js
//商品类目服务层
app.service("TbItemCatService",function($http){
	//查询所有
	this.findAll = function () {
		return $http.post("/TbItemCatController/findAll");
	}
	//查询实体
	this.findOne=function(id){
		return $http.post("/TbItemCatController/findOne?id="+id);
	}
	//根据父ID查询
	this.findByParentId = function (parentId) {
		return $http.post("/TbItemCatController/findByParentId?parentId=" + parentId);
	}
	//查询模板选项
	this.findTbTypeTemplateOptions=function () {
		return $http.post("/TbTypeTemplateController/findTbTypeTemplateOptions")
	}
	//增加、修改
	this.save=function (tbItemCat) {
		return $http.post("/TbItemCatController/save",tbItemCat);
	}
	//删除
	this.delete=function(ids){
		return $http.post("/TbItemCatController/delete?ids="+ids);
	}
});

File: dongyimai-interface/src/main/java/org/example/liuhengfei/service/TbItemCatService.java
package org.example.liuhengfei.service;

import org.example.liuhengfei.pojo.TbItemCat;

import java.util.List;

/**
 * 商品类目服务层接口
 *
 * @author Administrator
 */
public interface TbItemCatService {

    /**
     * 获取所有
     *
     * @return
     */
    List<TbItemCat> findAll();

    /**
     * 根据ID获取实体
     *
     * @param id
     * @return
     */
    TbItemCat findOne(Long id);

    /**
     * 增加、修改
     *
     * @param tbItemCat
     */
    void save(TbItemCat tbItemCat);

    /**
     * 批量删除
     *
     * @param ids
     */
    void delete(Long[] ids);

    /**
     * 根据父ID查询列表
     *
     * @param parentId
     * @return
     */
    List<TbItemCat> findByParentId(Long parentId);

}
